text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,411 @@
+
+Labels are mapped to a set of dedicated column scoped by the logical struct 'labels'. The list of labels can be easily
+determined from the schema by any participants. Metrics follow the same organization and are scoped by the logical
+struct 'metrics'.
+
+Attributes and exemplars are encoded with a list of structs instead of a struct of fields (columns). This mapping is
+based on the assumption that the number of attributes or exemplars can vary greatly from one measurement point to another
+for the same time-series. Arrow encodes this type representation with an offsets buffer and a child/data array.
+If this assumption is not true, another option will be to declare one column per attribute and exemplar with a validity
+bitmap enabled. **This should be validated by experimentation on realistic datasets.**
+
+For more details on the Arrow Memory Layout see this [document](https://arrow.apache.org/docs/format/Columnar.html#variable-size-binary-layout).
+
+```rust
+// Multivariate time-series schema declaration
+Schema::new(vec![
+    // Time range
+    Field::new("start_time_unix_nano", DataType::UInt64, false),
+    Field::new("time_unix_nano", DataType::UInt64, false),
+
+    // Labels
+    Field::new(
+            "labels",
+            DataType::Struct(vec![
+                Field::new("label_1", DataType::Utf8, false),
+                Field::new("label_2", DataType::Utf8, false),
+                // ...
+            ]),
+            true,
+    ),
+
+    // Metrics
+    // Only one metric in the structure if this schema represent a univariate time-series.
+    // Multiple fields in the structure to represent multivariate time-series.
+    Field::new(
+        "metrics",
+        DataType::Struct(vec![
+            Field::new("metric_1", DataType::Int64, false),
+            Field::new("metric_2", DataType::Float64, false),
+            // ...
+        ]),
+        false,
+    ),
+
+    // Attributes
+    Field::new("attributes", DataType::List(
+        Box::new(Field::new(
+            "attribute",
+            DataType::Struct(vec![
+                Field::new("name", DataType::Utf8, false),
+                Field::new("value", DataType::Utf8, false),
+            ]),
+            true,
+        ))
+    ), true),
+
+    // Exemplars
+    Field::new("exemplars", DataType::List(
+        Box::new(Field::new(
+            "exemplar",
+            DataType::Struct(vec![
+                Field::new("filtered_attributes", DataType::List(
+                    Box::new(Field::new(
+                        "attribute",
+                        DataType::Struct(vec![
+                            Field::new("name", DataType::Utf8, false),
+                            Field::new("value", DataType::Utf8, false),
+                        ]),
+                        true,
+                    ))
+                ), true),
+                Field::new("filtered_labels", DataType::List(
```

Comment:
Similar to @jmacd's comments on labels above, this should just be filtered_attributes

Reply:
Replaced labels by attributes.